src/Tutor.Core/LearnerModel/DomainOverlay/KnowledgeComponentMasteries/AssessmentItemMastery.cs

```diff
@@ -1,25 +1,64 @@
-﻿using System;
-using Tutor.Core.BuildingBlocks;
+﻿using FluentResults;
+using System;
+using Tutor.Core.BuildingBlocks.EventSourcing;
+using Tutor.Core.DomainModel.AssessmentItems;
 using Tutor.Core.LearnerModel.DomainOverlay.KnowledgeComponentMasteries.Events.AssessmentItemEvents;
 
 namespace Tutor.Core.LearnerModel.DomainOverlay.KnowledgeComponentMasteries
 {
-    public class AssessmentItemMastery: Entity
+    public class AssessmentItemMastery : EventSourcedEntity
     {
+        private const double PassThreshold = 0.9;
+
         public int AssessmentItemId { get; private set; }
         public double Mastery { get; private set; }
         public int SubmissionCount { get; private set; }
         public DateTime? LastSubmissionTime { get; set; }
-        public void UpdateMastery(AssessmentItemAnswered answer)
+        public bool IsCompleted { get => SubmissionCount > 0; }
+        public bool IsPassed { get => Mastery > PassThreshold; }
+
+        public void Select()
         {
-            if (Mastery <= answer.Evaluation.CorrectnessLevel) Mastery = answer.Evaluation.CorrectnessLevel;
-            SubmissionCount++;
-            LastSubmissionTime = answer.TimeStamp;
+            Causes(new AssessmentItemSelected()
+            {
+                AssessmentItemId = AssessmentItemId,
+            });
+        }
+
+        public void SubmitAnswer(Submission submission, Evaluation evaluation)
+        {
+            Causes(new AssessmentItemAnswered
+            {
+                AssessmentItemId = AssessmentItemId,
+                Submission = submission,
+                Evaluation = evaluation
+            });
+        }
+
+        public Result SeekHints()
+        {
+            // TODO: validation logic
```

Comment:
Remove the TODOs

Reply:
Removed in dda0d5278a8b594d2a8025b1344dc124ca7ddbb2.